packages/picasso/src/SidebarItemContent/SidebarItemContent.tsx

```diff
@@ -0,0 +1,128 @@
+import cx from 'classnames'
+import { makeStyles, Theme } from '@material-ui/core'
+
+import Typography from '../Typography'
+import Tooltip from '../Tooltip'
+import Container from '../Container'
+import Badge, { BadgeProps } from '../Badge'
+import { getNodeTextContent } from '../utils'
+import styles from './styles'
+
+export interface Props {
+  compact?: boolean
+  icon?: ReactElement
+  badgeProps?: Omit<BadgeProps, 'size' | 'children' | 'variant'>
+  children?: ReactNode
+  titleCase?: boolean
+  testIds?: {
+    content?: string
+  }
+}
+
+const useStyles = makeStyles<Theme>(styles, {
+  name: 'PicassoSidebarItemContent'
+})
+
+const resolveChildrenText = (text: ReactNode, titleCase: boolean) =>
+  typeof text === 'string' ? (
+    <Typography color='inherit' size='medium' titleCase={titleCase} noWrap>
+      {text}
+    </Typography>
+  ) : (
+    text
+  )
+
+interface ItemContentBadgeProps {
+  content: number
+  children?: ReactNode
+}
+
+const ItemContentBadge = (props: ItemContentBadgeProps) => {
+  const { content, children } = props
+  const classes = useStyles()
+  const isOverlay = React.Children.count(children) > 0
+
+  return (
+    <Badge
+      className={cx({
+        [classes.staticBadge]: !isOverlay
+      })}
+      content={content}
+      variant='red'
+      size={isOverlay ? 'small' : 'large'}
+    >
+      {children}
+    </Badge>
+  )
+}
+
+const CompactItemContent = (props: Props) => {
+  const { icon, children, badgeProps } = props
+  const classes = useStyles()
+
+  const hasBadge = badgeProps != null
+
+  let wrappedIcon = icon
+
+  if (hasBadge) {
+    wrappedIcon = (
+      <ItemContentBadge content={badgeProps.content}>
+        {wrappedIcon}
+      </ItemContentBadge>
+    )
+  }
+
+  return (
+    <Container className={classes.noWrap} inline flex alignItems='center'>
+      <Tooltip compact placement='right' content={getNodeTextContent(children)}>
+        <div>{wrappedIcon}</div>
+      </Tooltip>
+    </Container>
+  )
+}
+
+const ExpandedItemContent = (props: Props) => {
+  const { icon, badgeProps, children, testIds } = props
+  const classes = useStyles()
+
+  const hasIcon = icon != null
+  const hasBadge = badgeProps != null
+
+  return (
+    <Container className={classes.noWrap} inline flex alignItems='center'>
+      {icon}
+
+      <Container
+        className={cx(classes.noWrap, {
+          [classes.withIcon]: hasIcon
+        })}
+        flex
+        alignItems='center'
+        data-testid={testIds?.content}
+      >
+        {children}
+
+        {hasBadge && <ItemContentBadge content={badgeProps.content} />}
+      </Container>
+    </Container>
+  )
+}
+
+const SidebarItemContent = (props: Props) => {
+  const { children, titleCase: propsTitleCase, compact } = props
+
+  const titleCase = useTitleCase(propsTitleCase)
+  const resolvedChildren = resolveChildrenText(children, !!titleCase)
+
+  const ItemContentVariant = compact ? CompactItemContent : ExpandedItemContent
```

Comment:
nit: as soon as we don't export those components:

```suggestion
  const ItemContentVariant = compact ? Compact : Expanded
```